content/04-git-command-line/12-pull-upstream-changes.en.md

```diff
@@ -0,0 +1,65 @@
+---
+title: Pulling Upstream Changes
+weight: 12
+---
+
+Now that everyone has contributed to the [**RSE-Sheffield** repository](https://github.com/RSE-Sheffield/collaborative_github_exercise) (the **upstream** repository), the `params/` folder contains a number of files, one for each successful pull request.
+
+However your local copy of the repository only contains the template and your own file.
+
+> Q: How can I merge changes from the **upstream** repository to my **local** repository?
+
+
+### Add the **Upstream** Remote repository
+
+First, we have to configure `git` to know where our **upstream** repository is located by running
+```
+$ git remote add upstream https://github.com/RSE-Sheffield/collaborative_github_exercise.git
+```
+to add a remote repository named `upstream` at the URL `https://github.com/RSE-Sheffield/collaborative_github_exercise.git`.
+We can check that this worked using the following
+```
+$ git remote -v
+
+origin	git@github.com:johndoe/collaborative_github_exercise.git (fetch)
+origin	git@github.com:johndoe/collaborative_github_exercise.git (push)
+upstream	git@github.com:RSE-Sheffield/collaborative_github_exercise.git (fetch)
+upstream	git@github.com:RSE-Sheffield/collaborative_github_exercise.git (push)
+```
+This tells us the names of the remote repositories we have configured `origin` and `upstream` and their URLs.
+
+
+### Pull upstream changes
+
+To pull the upstream changes from the **`RSE-Sheffield`** repository, run the following commands
+```
+$ git checkout master
+$ git fetch upstream
```

Comment:
I prefer `git fetch --prune --all`, which will fetch from _all_ configured remotes (and `--prune` removes any unused tracking branches).